spec/jasmine-helper.js

```diff
@@ -0,0 +1,222 @@
+   * - If result is a JasmineNestedResults, it will be passed to rollupCounts().
+   * - If result is a JasmineExpectationResult, it will update this JasmineNestedResults's internal
+   *   counts.
+   * - If result is a JasmineMessageResult, it just gets added to the list of things returned by
+   *   getItems().
+   */
+  addResult(result: JasmineExpectationResult | JasmineNestedResults | JasmineMessageResult): void,
+  passed(): boolean,
+},
+
+/**
+ * This interface is defined by https://github.com/jasmine/jasmine/blob/1_3_x/src/core/Runner.js.
+ */
+type JasmineRunner = {
+  constructor(env: JasmineEnv): void,
+  results(): JasmineNestedResults,
+  specs(): JasmineSpec[],
+  topLevelSuites(): JasmineSuite[],
+
+  // Admittedly, this has many more methods, but implementations of JasmineReporter that make a
+  // JasmineRunner available should primarily be concerned about the observer methods.
+}
+
+/**
+ * This interface is defined by https://github.com/jasmine/jasmine/blob/1_3_x/src/core/Spec.js.
+ */
+type JasmineSpec = {
+  constructor(env: mixed, suite: JasmineSuite, description: string): void,
+  getFullName(): string,
+  results(): JasmineNestedResults,
+}
+
+/**
+ * This interface is defined by https://github.com/jasmine/jasmine/blob/1_3_x/src/core/Suite.js.
+ */
+type JasmineSuite = {
+  constructor(
+    env: JasmineEnv,
+    description: string,
+    specDefinitions: Function,
+    parentSuite: JasmineSuite,
+  ): void,
+  getFullName(): string,
+  finish(onComplete: () => void): void,
+  add(suiteOrSpec: JasmineSuite | JasmineSpec): void,
+  specs(): JasmineSpec[],
+  suites(): JasmineSuite[],
+  children(): Array<JasmineSuite | JasmineSpec>,
+  execute(onComplete: () => void): void,
+}
+
+/**
+ * This interface is defined by https://github.com/jasmine/jasmine/blob/1_3_x/src/core/Reporter.js.
+ */
+type JasmineReporter = {
+  /** This is invoked once when the runner starts up. */
+  reportRunnerStarting(runner: JasmineRunner): void,
+
+  /**
+   * This will be invoked when the runner is completely finished. At this point, it makes sense to
+   * inspect the runner.
+   */
+  reportRunnerResults(runner: JasmineRunner): void,
+
+  /** As an example, TerminalReporter does not use this method. */
+  reportSuiteResults(suite: JasmineSuite): void,
+
+  /** As an example, TerminalReporter does not use this method. */
+  reportSpecStarting(spec: JasmineSpec): void,
+
+  /** This is frequently used to report streaming results. */
+  reportSpecResults(spec: JasmineSpec): void,
+
+  /** Reports a log message to print out. */
+  log(str: string): void,
+}
+
+/**
+ * @param specSuite must be a path to a file that, when required, exports a function that takes no
+ *   arguments and returns a new object that satisfies the contract of `JasmineEnv`.
+ * @param logFile If specified, log messages from the test will be written to this file; otherwise,
+ *   they will be written to stderr. (This only applies when tests are run from the terminal.)
```

Comment:
@bolinfest Do you know if there's any documentation-generator tool that uses flow's type annotations to linkify things in the way that Atom's docs system does?